miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/userinfo/UserInfoBridge.kt

```diff
@@ -92,28 +96,43 @@ internal class UserInfoBridge {
 
     private fun onHasAccessTokenPermission(callbackObj: CallbackObj) {
         val tokenPermission = parseAccessTokenPermission(callbackObj)
+        if (doesMatchManifest(tokenPermission)) {
 
-        val successCallback = { accessToken: TokenData ->
-            bridgeExecutor.postValue(callbackObj.id, Gson().toJson(accessToken))
-        }
-        val errorCallback = { message: String ->
-            bridgeExecutor.postError(callbackObj.id, "$ERR_GET_ACCESS_TOKEN $message")
-        }
+            val successCallback = { accessToken: TokenData ->
+                accessToken.accessTokenScope = tokenPermission
+                bridgeExecutor.postValue(callbackObj.id, Gson().toJson(accessToken))
+            }
+            val errorCallback = { message: String ->
+                bridgeExecutor.postError(callbackObj.id, "$ERR_GET_ACCESS_TOKEN $message")
+            }
 
-        try {
-            userInfoBridgeDispatcher.getAccessToken(miniAppId, tokenPermission, successCallback, errorCallback)
-        } catch (e: MiniAppSdkException) {
-            if (e.message == "The `UserInfoBridgeDispatcher.getAccessToken` $NO_IMPL")
-                userInfoBridgeDispatcher.getAccessToken(miniAppId, successCallback, errorCallback)
-            else
-                throw e
+            try {
+                userInfoBridgeDispatcher.getAccessToken(miniAppId, tokenPermission, successCallback, errorCallback)
+            } catch (e: MiniAppSdkException) {
+                if (e.message == "The `UserInfoBridgeDispatcher.getAccessToken` $NO_IMPL")
+                    userInfoBridgeDispatcher.getAccessToken(miniAppId, successCallback, errorCallback)
+                else
+                    throw e
+            }
+        } else
+            bridgeExecutor.postError(callbackObj.id, "$ERR_GET_ACCESS_TOKEN $ERR_ACCESS_TOKEN_NOT_MATCH_MANIFEST")
+    }
+
+    private fun doesMatchManifest(tokenScope: AccessTokenScope): Boolean {
+        if (tokenScope.scopes.isNotEmpty()) {
+            val atpList = downloadedManifestCache.getAccessTokenPermissions(miniAppId)
+            atpList.forEach {
+                if (it.audience == tokenScope.audience && it.scopes.containsAll(tokenScope.scopes))
+                    return true
```

Comment:
Using `forEach { }` will let you to walk through every iteration, I have a feelings that we'll be lucky if we find the match at the first iteration and unlucky when it will find the match at the last. 

Instead we may do something using `find { }` e.g.

```
val cachedAccessTokens = downloadedManifestCache.getAccessTokenPermissions(miniAppId)
cachedAccessTokens.find {
   it.audience == tokenScope.audience && it.scopes.containsAll(tokenScope.scopes)
}?.let { return true }
```

But, anyway this is not the ideal time for tracking the performance issue, so you can consider this comment as optional.

Reply:
it's not true, when you're using `find { }` then there is the possibility that it must go through every element in the list to get an item that you want. Furthermore, when you use find then you return an item value, it's a redundant step.

I can change into a normal loop if it makes you feel less iterated action.

Reply:
Ok, if that so, I think, you don't need to change here anything for now.